wimm.Secundatives/Result.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using wimm.Secundatives.Extensions;
+
+namespace wimm.Secundatives
+{
+    /// <summary>
+    /// A basic Result class that handles operations with a possibility of failure
+    /// </summary>
+    /// <typeparam name="T"> The type to be returned on success </typeparam>
+    public class Result<T> : Variant<T,Error>
+    {
+
+        /// <summary> A value indicating whether the operation ended in error </summary>
+        public bool IsError => Is<Error>();
+
+        /// <summary> A value indicating if the operation successfully returned a value </summary>
+        public bool IsSome => Is<T>();
```

Comment:
This one doesn't seem strictly necessary, since it already has `Is<T>` from the base and `IsError` is known to be the non-value type. When I saw the name in the test I assumed it did something else, maybe it threw me because I thought of "some" as opposed to "just one", rather than as opposed to Error